### DISCUSSION TITLE
Node callbacks

### DISCUSSION THREAD

**Discussion by arturolinares** [slint-ui/slint#2065] [Q&A]
Hi! I am having trouble using the slint-ui with node in a small application.

I have a window with a button that when it is clicked makes a request to Google and updates the button text to the response body length. The UI appears to work well, except that the request doesn't seem to finish until the window is closed. Maybe I need to perform the request outside the UI thread... but how can I do that? 

![Peek 13-01-2023 19-45](https://user-images.githubusercontent.com/203078/212445021-87a5f512-ef1b-4339-8db2-58083f8030bd.gif)

This is the script:

```js
const slint = require("slint-ui");
const ui = require("./ui/main.slint");

const fetch = require('node-fetch');

function doFetch() {
	fetch('https://google.com')
		.then(r => r.text())
		.then(text => {
			console.log("Update button text");
			app.MainButton.text = text.length;
		});
}

let app = new ui.MainWindow({
	clicked: () => doFetch()
});
app.run();
```

The slint file:
```slint
import { Button } from "std-widgets.slint";

export MainWindow := Window {
	callback clicked <=> MainButton.clicked;
	MainButton := Button {
		text: "Click Me!";
	}
}
```

Thanks in advance :smile:

**Comment by tronical** [comment]
I think I can see the problem, we're not waking up the libuv event loop from within our event loop :(. I don't know yet how to fix that.

**Reply by tronical** [reply]
Thinking out loud: Afaics we could run the winit loop on another thread everywhere except macOS, according to the winit API. For macOS if libuv is using cfrunloop then perhaps we can reach out to that directly.

**Reply by arturolinares** [reply]
Oh no! v1.0.0 was released and this was not fixed! Should I open an issue?

**Reply by tronical** [reply]
Yes, that sounds like a good idea. Actually, I think I can just convert this discussion into an issue :)

**Reply by tronical** [reply]
Created #2477
